probes/hasApprovedLicense/def.yml

```diff
@@ -0,0 +1,27 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: hasApprovedLicense
+short: Check that the project has an FSF or OSI approved license. For more, see [all FSF or OSI approved license formats](https://spdx.org/licenses/).
+motivation: >
+  A license can give users information about how the source code may or may not be used. The lack of a license will impede any kind of security review or audit and creates a legal risk for potential users.
+implementation: >
+ The implementation checks whether a license file is present and is of an approved format
+outcome:
+  - If a license file is found and is of an approved format, the probe returns a single OutcomePositive (1).
+  - If a license file is missing or is not of an approved format, the probe returns a single OutcomeNegative (0).
```

Comment:
if the license is missing should this be OutcomeNotApplicable? I could see either outcome being fine, just a discussion in case it's relevant.

Reply:
Done in https://github.com/ossf/scorecard/pull/3465/commits/70c4f465a5212d2edeb71eb834667f4a115c344a